WordPress/Classes/ViewRelated/Blog/SettingsTextViewController.m

```diff
@@ -0,0 +1,92 @@
+#import "SettingsTextViewController.h"
+#import "WPTextFieldTableViewCell.h"
+#import "WPStyleGuide.h"
+
+static NSString * const SiteTitleTextCell = @"SiteTitleTextCell";
+
+@interface SettingsTextViewController()
+
+@property (nonatomic, strong) WPTextFieldTableViewCell *textFieldCell;
+@property (nonatomic, strong) NSString *hint;
+
+@end
+
+@implementation SettingsTextViewController
+
+- (instancetype)initWithText:(NSString *)text placeholder:(NSString *)placeholder hint:(NSString *)hint isPassword:(BOOL)isPassword
+{
+    self = [super initWithStyle:UITableViewStyleGrouped];
+    if (self) {
+        _textFieldCell = [[WPTextFieldTableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:SiteTitleTextCell];
+        _textFieldCell.textField.clearButtonMode = UITextFieldViewModeAlways;
+        _textFieldCell.minimumLabelWidth = 0.0f;
+        [WPStyleGuide configureTableViewTextCell:_textFieldCell];
+        _textFieldCell.textField.text = text;
+        _textFieldCell.textField.placeholder = placeholder;
+        _textFieldCell.textField.returnKeyType = UIReturnKeyDone;
+        _textFieldCell.textField.keyboardType = UIKeyboardTypeASCIICapable;
+        [_textFieldCell.textField becomeFirstResponder];
+        _textFieldCell.shouldDismissOnReturn = YES;
+        _textFieldCell.textField.secureTextEntry = YES;
```

Comment:
We probably shouldn't be doing all this view initialization on `init`. I'd at least move it to `viewDidLoad`, or better add a `textFieldCell` method that initializes the cell if `_textFieldCell` is `nil`